src/Core/Types/Types/Relay/IdFieldValueSerializer.cs

```diff
@@ -102,6 +104,11 @@ public void Initialize(NameString schemaName)
                     .Build());
         }
 
+        protected virtual IdValue DeserializeId(string s)
+        {
+            return InnerSerializer.Deserialize(s);
+        }
```

Comment:
Potentially this could be moved behind an interface that gets passed into the ctor, of shape:
```csharp
IdValue DeserializeId(string value, Type valueType, string schemaName, string typeName);
```

Then in the Id extension methods class that instantiates this `IdFieldValueSerializer`, you could try to get from DI an implementation, else just fall back to a default w/ `InnerSerializer.Deserialize(s)`.

Then you wouldn't need the factory at all.